### PR TITLE
Register Util

### PR DESCRIPTION
**Pull request by namebee**

### CODE REVIEW

**Review by YTG1234 — APPROVED** [review]
Sure, I guess?

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * @param name
+     *        Identifier of item instance to be registered
+     * @param item
+     *        Item instance to be registered
+     * @return Item instance registered
+     */
+    public static <T extends Item> T registerItems(Identifier name, T item) {
```
Comment (dzwdz) [inline src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: registerItem**s**? it only registers one

**Review by dzwdz — COMMENTED** [review]
the function names could be shortened to eg `ProtonRegisterUtil.item(...)`, as the class name makes it obvious that they register the items anyways

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java (hunk 1 above)
Comment (YTG1234) [reply src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: ~~rename it to `registerItemWithIdentifier`~~

**Comment by YTG1234** [comment]
> the function names could be shortened to eg `ProtonRegisterUtil.item(...)`, as the class name makes it obvious that they register the items anyways

I agree

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    /**
+     * @param name
+     *        Name of block instance to be registered
+     * @param block
+     *        Block instance to be registered
+     * @param settings
+     *        Item.Settings of BlockItem of Block instance to be registered
+     * @return Block instance registered
+     */
+    public static <T extends Block> T blocks(String name, T block, Item.Settings settings) {
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: should be singular `block`

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    /**
+     * @param name
+     *        Name of block instance to be registered
+     * @param block
+     *        Block instance to be registered
+     * @param item
+     *        BlockItem instance of Block to be registered
+     * @return Block instance registered
+     */
+    public static <T extends Block> T blocks(String name, T block, BlockItem item) {
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: should be singular `block`

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * @param name
+     *        Name of block instance to be registered
+     * @param block
+     *        Block instance to be registered
+     * @return Block instance registered
+     */
+    public static <T extends Block> T blocks(String name, T block) {
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: should be singular `block`

File: src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * @param name
+     *        Identifier of block instance to be registered
+     * @param block
+     *        Block instance to be registered
+     * @return Block instance registered
+     */
+    public static <T extends Block> T blocks(Identifier name, T block) {
```
Comment (YTG1234) [inline src/main/java/io/github/hydos/proton/util/ProtonRegisterUtil.java]: should be singular `block`

**Review by YTG1234 — CHANGES_REQUESTED** [review]

**Comment by dzwdz** [comment]
that's a lot of javadocs

**Comment by YTG1234** [comment]
Javadoc is cool

**Review by dzwdz — APPROVED** [review]

**Pull request merged**